whois_rdap_connector.py

```diff
@@ -107,12 +128,12 @@ def _handle_test_connectivity(self, param):
 
         self.save_progress("Parsing response")
 
-        if whois_response["query"] and whois_response["query"] == ip:
+        if whois_response.get("query", None) and whois_response.get("query", None) == ip:
```

Comment:
Performing a 'get' will default to 'None' if the key is not found, no need to set it as default.

Reply:
Yes @tonyc-phantom, I have updated code.